src/test/java/org/folio/rest/api/PrecedingSucceedingTitleTest.java

```diff
@@ -1,43 +1,66 @@
+import static org.folio.rest.api.entities.PrecedingSucceedingTitle.SUCCEEDING_INSTANCE_ID_KEY;
+import static org.folio.rest.support.http.InterfaceUrls.precedingSucceedingTitleUrl;
+import static org.folio.utility.RestUtility.TENANT_ID;
+import static org.folio.utility.VertxUtility.getClient;
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.CoreMatchers.nullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.equalTo;
 
-import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
-import static org.folio.rest.api.entities.PrecedingSucceedingTitle.PRECEDING_INSTANCE_ID_KEY;
-import static org.folio.rest.api.entities.PrecedingSucceedingTitle.SUCCEEDING_INSTANCE_ID_KEY;
-import static org.folio.rest.support.http.InterfaceUrls.instancesStorageUrl;
-import static org.folio.rest.support.http.InterfaceUrls.precedingSucceedingTitleUrl;
-
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
 import java.net.HttpURLConnection;
 import java.util.List;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
-
+import lombok.SneakyThrows;
 import org.folio.rest.api.entities.PrecedingSucceedingTitle;
 import org.folio.rest.api.entities.PrecedingSucceedingTitles;
 import org.folio.rest.jaxrs.model.Errors;
 import org.folio.rest.support.IndividualResource;
 import org.folio.rest.support.Response;
 import org.folio.rest.support.ResponseHandler;
-
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 
-import io.netty.handler.codec.http.HttpResponseStatus;
-import io.vertx.core.json.JsonArray;
-import io.vertx.core.json.JsonObject;
-
-
 public class PrecedingSucceedingTitleTest extends TestBaseWithInventoryUtil {
   private static final String INVALID_UUID_ERROR_MESSAGE = "Invalid UUID format of id, should be " +
     "xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx where M is 1-5 and N is 8, 9, a, b, A or B and x is 0-9, a-f or A-F.";
   private static final String HRID = "inst000000000022";
   private static final String TITLE = "A web primer";
 
+  @SneakyThrows
+  @Before
+  public void beforeEach() {
+    StorageTestSuite.deleteAll(TENANT_ID, "preceding_succeeding_title");
+    StorageTestSuite.deleteAll(TENANT_ID, "instance_relationship");
+    StorageTestSuite.deleteAll(TENANT_ID, "bound_with_part");
+
+    deleteAllById(precedingSucceedingTitleClient);
+
+    clearData();
+    setupMaterialTypes();
+    setupLoanTypes();
+    setupLocations();
+    removeAllEvents();
+  }
+
+  @After
+  public void afterEach() {
+    StorageTestSuite.deleteAll(TENANT_ID, "preceding_succeeding_title");
+    StorageTestSuite.deleteAll(TENANT_ID, "instance_relationship");
+    StorageTestSuite.deleteAll(TENANT_ID, "bound_with_part");
+
+    deleteAllById(precedingSucceedingTitleClient);
+  }
```

Comment:
Why is this being done before and after each test?

Reply:
I remember this one giving me a lot of trouble with affecting tests from other classes.

I think I can try and AfterClass rather than an After each.
I am pretty sure I cannot remove the After/AfterClass in this particular case.

Reply:
What leads to a `before each` being insufficient to achieve this?

Reply:
The state After the class is done with is different.
Tests from other classes may fail as a result.
The existing of preceding_succeeding_title, instance_relationship, bound_with_part can cause deletes to fail due to constraint violations.
The lack of proper deletion results in an inconsistent state and then tests sometimes have the wrong number of rows.
Then test assertions fail.

Switching to an `@AfterClass` appears to still fix the state for everything outside of this class without running between every test.
We should be able to remove such logic if and once we get data isolation between test classes.

Reply:
Thanks for explaining the reasoning further.

Please add a comment to explain that reasoning.

I reckon this is partly because of the mixture of suite and non-suite tests and inconsistent `before` methods. The original intent was that the `before` would clean the state of the system (without enabling a new tenant because that is the most time consuming preparatory step).

Reply:
Comments added.